windows-agent/cmd/ubuntu-pro-agent/agent/agent.go

```diff
@@ -111,16 +125,50 @@ func (a *App) serve(args ...option) error {
 	}
 	defer proservice.Stop(ctx)
 
-	daemon, err := daemon.New(ctx, proservice.RegisterGRPCServices, daemon.WithAddrDir(opt.daemonAddrDir))
+	a.daemon = daemon.New(ctx, proservice.RegisterGRPCServices, publicDir)
+
+	close(a.ready)
+
+	return a.daemon.Serve(ctx)
+}
+
+func setupDirectories(ctx context.Context, opt *options) (public, private string, err error) {
+	public, err = vauleOrAfterEnv(opt.publicDir, "UserProfile", common.LocalAppDataDir)
 	if err != nil {
-		close(a.ready)
-		return err
+		return "", "", err
 	}
+	log.Debugf(ctx, "Agent public directory: %s", public)
 
-	a.daemon = &daemon
-	close(a.ready)
+	private, err = vauleOrAfterEnv(opt.privateDir, "LocalAppData", common.UserProfileDir)
+	if err != nil {
+		return "", "", err
+	}
+	log.Debugf(ctx, "Agent private directory: %s", private)
+
+	if err := os.MkdirAll(public, 0600); err != nil {
+		return "", "", fmt.Errorf("could not create public directory: %v", err)
+	}
+
+	if err := os.MkdirAll(private, 0600); err != nil {
+		return "", "", fmt.Errorf("could not create private directory: %v", err)
+	}
+
+	return public, private, nil
+}
+
+// vauleOrAfterEnv is a helper for parsing optional inputs.
+// Returns "intput" if it is non-empty. Otherwise it returns "${env}/relative".
+func vauleOrAfterEnv(input string, env string, relative string) (string, error) {
```

Comment:
```suggestion
// valueOrAfterEnv is a helper for parsing optional inputs.
// Returns "input" if it is non-empty. Otherwise it returns "${env}/relative".
func valueOrAfterEnv(input string, env string, relative string) (string, error) {
```